cow_instrument/testing/DetectorComponentTest.cpp

```diff
@@ -65,9 +65,117 @@ TEST(detector_component_test, test_copy) {
 TEST(detector_component_test, test_not_path_component) {
   DetectorComponent detector(ComponentIdType{1}, DetectorIdType{1},
                              V3D{1, 1, 1});
-  bool isPathComponent = std::is_base_of<PathComponent, DetectorComponent>::value;
+  bool isPathComponent =
+      std::is_base_of<PathComponent, DetectorComponent>::value;
   EXPECT_FALSE(isPathComponent)
       << "DetectorComponent should not have public base PathComponent";
 }
 
+TEST(detector_component_test, test_single_rotation_around_detector_origin) {
+
+  const Eigen::Vector3d rotationAxis{0, 0, 1};
+  const double rotationAngle = M_PI / 2;
+  const Eigen::Vector3d rotationCenter{0, 0, 0};
+
+  DetectorComponent detector(
+      ComponentIdType(1), DetectorIdType{1},
+      rotationCenter /*I make this the position of the detector*/);
+  detector.rotate(rotationAxis, rotationAngle, rotationCenter);
+  // Check that the position has the identity rotation applied.
+  EXPECT_TRUE(detector.getPos().isApprox(rotationCenter, 1e-14))
+      << "Position should not change";
+
+  // Check that the internal rotation gets updated. i.e component is rotated
+  // around its own centre.
+  Eigen::Matrix3d rotMatrix = detector.getRotation().toRotationMatrix();
+  // Check that some vector I define gets rotated as I would expect
+  Eigen::Vector3d rotatedVector = rotMatrix * Eigen::Vector3d{1, 0, 0};
+  EXPECT_TRUE(rotatedVector.isApprox(Eigen::Vector3d{0, 1, 0}, 1e-14))
+      << "Internal detector rotation not updated correctly";
+}
+
+TEST(detector_component_test, test_multiple_rotation_around_detector_origin) {
+
+  const Eigen::Vector3d rotationAxis{0, 0, 1};
+  const double rotationAngle = M_PI / 2;
+  const Eigen::Vector3d rotationCenter{0, 0, 0};
+
+  DetectorComponent detector(
+      ComponentIdType(1), DetectorIdType{1},
+      rotationCenter /*I make this the position of the detector*/);
+
+  // Rotate once by 90 degrees
+  detector.rotate(rotationAxis, rotationAngle, rotationCenter);
+  // Rotate again by 90 degrees
+  detector.rotate(rotationAxis, rotationAngle, rotationCenter);
```

Comment:
We should have tests here that test two additional things:

- The **ordering** of rotations is correct
- The **sign** of rotations is correct

With the parameters chosen here the test would not fail of any or both were wrong.